src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4465,217 @@ subroutine cat_enkf_increments(                               &
+       
+       ! get target for snow layer thickness (as used in subroutine catchment() over land tiles)
+       
+       targetthick(1)        = CATCH_DZ1MAX
+       
+       targetthick(2:N_snow) = 1./(N_snow-1.)
+       
+       ! identify the obs species of interest       
+       
+       N_select_varnames  = 1      
+       
+       select_varnames(1) = 'asnow'
+       
+       call get_select_species(                                           &
+            N_select_varnames, select_varnames(1:N_select_varnames),      &
+            N_obs_param, obs_param, N_select_species, select_species )
+       
+       allocate(select_tilenum(1))
+       
+       swe_incr = 0.   ! total SWE increment; initialize to NO CHANGE
+       
+       ! loop through tiles and compute increments
+       
+       do kk=1,N_catd
+          
+          ! find observations for tile kk
+          
+          select_tilenum(1) = l2f(kk)
+          
+          call get_ind_obs(                                           &
+               N_obs,            Observations,                        &
+               1,                select_tilenum,                      &
+               N_select_species, select_species(1:N_select_species),  &
+               N_selected_obs,   ind_obs )
+          
+          if (N_selected_obs > 0) then
+             
+             ! average in case there are multiple "asnow" obs (e.g., from MODIS and VIIRS)
+             
+             tmp_obs = sum(Observations(ind_obs(1:N_selected_obs))%obs)
+             
+             if (N_selected_obs > 1)  tmp_obs = tmp_obs/real(N_selected_obs)
+             
+             do n_e=1,N_ens  ! compute analysis separately for each ensemble member
+                
+                ! 1. Diagnose model forecast snow cover area fraction and total SWE
+                
+                asnow_fcst  = asnow(kk,n_e)
+                swe_fcst    = sum(cat_progn(kk,n_e)%wesn(1:N_snow))
+                
+                ! 2. Calculate SWE increment based on modified eq 1 of Toure et al (2018)
+                
+                if     (asnow_fcst .lt. tmp_obs * SCF_ANA_ALPHA) then
+                   
+                   ! ADD SNOW:    Forecast SCF is less than observed SCF (after "bias" adjustment with alpha)
+                   
+                   swe_incr(kk,n_e) = SCF_ANA_MAXINCRSWE * (tmp_obs - asnow_fcst/SCF_ANA_ALPHA)
+                   
+                elseif (tmp_obs .lt. SCF_ANA_BETA) then
+                   
+                   ! REMOVE SNOW: Simulated SCF is greater than observed SCF (after "bias" adjustment)
+                   !                and observed SCF is less than beta threshold
+                   
+                   swe_incr(kk,n_e) = (-1.) *  SCF_ANA_MAXINCRSWE * asnow_fcst * (1. - tmp_obs/SCF_ANA_BETA)
+                   
+                else 
+                   
+                   cycle  ! NO CHANGE, skip rest of increment calcs and go straight to next ens member 
```

Comment:
Note new "cycle" statement here.  I think this is right, but if the revised code gives very different results, this change should be examined further